src/api/app/controllers/webui/package_controller.rb

```diff
@@ -705,28 +705,29 @@ def check_build_log_access
       return false
     end
 
+    @package_name = params[:package]
     begin
-      @package = Package.get_by_project_and_name(@project, params[:package], use_source: false,
-                                                                             follow_multibuild: true,
-                                                                             follow_project_links: true)
+      @package = Package.get_by_project_and_name(@project, @package_name, use_source: false,
+                                                                          follow_multibuild: true,
+                                                                          follow_project_links: true)
     rescue Package::UnknownObjectError
       redirect_to project_show_path(@project.to_param),
                   error: "Couldn't find package '#{params[:package]}' in " \
                          "project '#{@project.to_param}'. Are you sure it exists?"
       return false
     end
 
-    # package is nil for remote projects
+    # NOTE: @package is nil for remote projects
     if @package && !@package.check_source_access?
-      redirect_to package_show_path(project: @project.name, package: @package.name),
+      redirect_to package_show_path(project: @project.name, package: @package_name),
                   error: 'Could not access build log'
       return false
     end
 
     @can_modify = User.possibly_nobody.can_modify?(@project) || User.possibly_nobody.can_modify?(@package)
 
-    # for remote and multibuild / local link packages
-    @package = params[:package] if @package.try(:name) != params[:package]
+    # NOTE: @package is a String for multibuild packages
```

Comment:
And for remote packages, right?

Reply:
For remote packages it is an object, yes.

But for local link packages `@package` is nil, so the code I changed breaks.

I will change the PR to draft until I fix this and add tests.